From 6f9d61569750fd6e3a5dd78985fea20e069c1a1d Mon Sep 17 00:00:00 2001
From: Andre Meyering <info@andremeyering.de>
Date: Sat, 29 May 2021 13:41:44 +0200
Subject: [PATCH] [QuaZip] Use header include path "quazip/" instead of "quazip5/"

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fe2d8d168..ecde2c5a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -206,6 +206,24 @@ configure_file(data/i18n.qrc ${CMAKE_BINARY_DIR} COPYONLY)
 
 add_subdirectory(docs)
 add_subdirectory(third_party EXCLUDE_FROM_ALL)
+
+if(USE_EXTERN_QUAZIP)
+  find_package(QuaZip-Qt5 QUIET)
+
+  if(TARGET QuaZip::QuaZip)
+    message(STATUS "Using system's QuaZip 1.x library")
+    # We use this alias to still support QuaZip 0.9 which doesn't provide a
+    # CMake config file which means we can't use find_package() on it.
+    add_library(quazip5 ALIAS QuaZip::QuaZip)
+  else()
+    # The system's quazip, e.g. libquazip5-dev on Ubuntu
+    message(
+      STATUS
+        "Using system's QuaZip 0.9 library; expecting quazip5 shared library to exist"
+    )
+  endif()
+endif()
+
 add_subdirectory(src)
 
 add_executable(
diff --git a/src/export/ExportTemplateLoader.cpp b/src/export/ExportTemplateLoader.cpp
index fec2ec1ae..b562bfa30 100644
--- a/src/export/ExportTemplateLoader.cpp
+++ b/src/export/ExportTemplateLoader.cpp
@@ -6,13 +6,8 @@
 #include <QNetworkReply>
 #include <QXmlStreamReader>
 
-#ifndef EXTERN_QUAZIP
-#    include "quazip/quazip/quazip.h"
-#    include "quazip/quazip/quazipfile.h"
-#else
-#    include "quazip5/quazip.h"
-#    include "quazip5/quazipfile.h"
-#endif
+#include "quazip/quazip.h"
+#include "quazip/quazipfile.h"
 
 #include "globals/VersionInfo.h"
 #include "log/Log.h"
diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index df74a5fb9..641b7094f 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -1,14 +1,9 @@
-if(USE_EXTERN_QUAZIP)
-  # The system's quazip, e.g. libquazip5-dev on Ubuntu
-  message(STATUS "Using system's QuaZip library")
-  add_library(quazip5 SHARED IMPORTED)
-
-else()
+if(NOT USE_EXTERN_QUAZIP)
   find_package(ZLIB REQUIRED)
 
   add_library(quazip5 STATIC)
 
-  target_include_directories(quazip5 SYSTEM INTERFACE .)
+  target_include_directories(quazip5 SYSTEM INTERFACE quazip)
   target_include_directories(quazip5 PRIVATE quazip/quazip)
   target_compile_definitions(quazip5 PRIVATE QUAZIP_BUILD)
   target_compile_options(quazip5 PRIVATE -w) # Disable warnings for third-party
